galaxy/main/migrations/0142_collection_search.py

```diff
@@ -0,0 +1,82 @@
+from django.contrib.postgres import fields as psql_fields
+from django.db import migrations
+
+
+TS_VECTOR_SELECT = '''
+    setweight(to_tsvector(coalesce(cv.metadata ->> 'namespace', '')), 'A')
+    || setweight(to_tsvector(coalesce(cv.metadata ->> 'name', '')), 'A')
+    || (
+      SELECT
+        setweight(to_tsvector(
+          coalesce(string_agg(cvt, ' '), '')
+        ), 'B')
+      FROM jsonb_array_elements_text(cv.metadata -> 'tags') cvt
+    )
+    || (
+      SELECT
+        setweight(to_tsvector(
+          coalesce(string_agg(cvc ->> 'name', ' '), '')
+        ), 'C')
+      FROM jsonb_array_elements(cv.contents) AS cvc
+    )
+    || setweight(to_tsvector(
+         coalesce(cv.metadata ->> 'description', '')
+       ), 'D')
+'''
```

Comment:
Please document what this query will be doing.

Reply:
Will do.